src/mds/Server.cc

```diff
@@ -2139,7 +2197,53 @@ void Server::dispatch_client_request(MDRequestRef& mdr)
 
   if (mdr->killed) {
     dout(10) << "request " << *mdr << " was killed" << dendl;
-    return;
+    //if the mdr is a "batch_op" and it has followers, pick a follower as
+    //the new "batch_head" and go on processing the new one.
+    if (mdr->is_batch_op() && mdr->is_batch_head && mdr->batch_reqs.size()) {
+      MDRequestRef new_batch_head;
+      for (auto req : mdr->batch_reqs) {
+	mdr->batch_reqs.erase(req);
+	if (!req->killed) {
+	  new_batch_head = req;
+	  break;
+	}
+      }
+
+      if (!new_batch_head) 
+	return;
+
+      for (auto r : mdr->batch_reqs) {
+	r->batch_head = new_batch_head;
+	new_batch_head->batch_reqs.insert(r);
+	mdr->batch_reqs.erase(r);
+      }
```

Comment:
you can't use erase() inside the loop. besides you shouldn't add new_batch_head itself to batch_reqs

Reply:
@ukernel Hi:-)
new_batch_head wouldn't be added to batch_reqs, as they are already removed from the mds->batch_reqs in the former loop:-)

Reply:
@ukernel 
I've corrected the erase() issue, please take look:-) Thanks